### PR TITLE
Refactor settings to their own tabs and routes

### PR DESCRIPTION
**Pull request by MaxLeiter**
~This seems pretty controversial from some discussion on IRC, but I like it so I'll leave it up for discussion here before closing it.~

See https://github.com/thelounge/thelounge/pull/4489#issuecomment-1047241161 for updated screenshots

|screenshots:||
|--------------|:-----:|
| image | image
| image |  image

I also split each tab into its own component and route, as the Settings.vue file is 600+ lines long. 

While the settings aren't too complex right now, this allows us more flexibility to add more without worrying about clutter + we can add more tabs for features like custom keybinds.

### CODE REVIEW

**Comment by itsjohncs** [comment]
I like the use of text + icons much more than just icons. It is less aesthetically pleasing but (I think) much more useable. We can do some more passes over the aesthetics after landing too.

As I said on IRC, I don't think the settings page has enough in it to benefit from this. Since separating them out into separate pages that you need to explore makes them slower to look through I think this isn't a net positive. On a meaty desktop or tablet screen I feel like this is particularly true.

I'm also hesitant to encourage the proliferation of settings since I think each additional setting comes with drawbacks that need to be weighed.

But all that said, I don't really have a big problem with this change and it seemed like a few people were excited about it at least. So I won't fuss if people generally want to go through with it.

Making all the pages visible at once (perhaps allowing quick jumping between them using a sidebar) on large screens would alleviate my first concern too. That could be done after this change lands easily enough.

I can do a proper review after we arrive at consensus about what we're doing, but some quick thoughts I wanted to jot down so I don't lose them:

* Autocomplete doesn't seem like it belongs under appearance.
* "Account" might be a more widely-intuitive category title than "User Settings".

**Comment by MaxLeiter** [comment]
> ...separating them out into separate pages that you need to explore makes them slower to look through I think this isn't a net positive

This is true if you know what you're looking for, but I'd argue the tabbed system is newer for easier users, especially with your Account suggestion. Four options is not that many to consider, I think it's pretty obvious what setting should be where.

> I'm also hesitant to encourage the proliferation of settings since I think each additional setting comes with drawbacks that need to be weighed.

Agreed, but I do think there are valuable options this allows us to now explore, like admin settings, upload management, etc. 

> Autocomplete doesn't seem like it belongs under appearance.

This isn't really new -- autocomplete has always been under the "Visual aids" section. Where do you think it should be instead? General?

> "Account" might be a more widely-intuitive category title than "User Settings".

Thanks, I was struggling to think of a better name last night 😅

**Comment by MaxLeiter** [comment]
screenshot showing the new 'account' setting instead of 'user settings'

**Comment by MaxLeiter** [comment]
Worth pointing out that #85 and @astorije's mock-up of better settings exists at https://fishmonger-christopher-73101.netlify.app/.

**Comment by GewoonYorick** [comment]
Very welcome change, Opens up the door to a lot of new possibilities and gets rid of long list of settings.

Personally I feel like it'd be nicer to have the navigator on the side (and not ontop) on wider screens atleast.

Currently the navigator (selected item) looks a little off on the default theme: 
image

**Comment by MaxLeiter** [comment]
@GewoonYorick I hadn't tested on non-morning yet, oops. Thanks!

I'll play around with larger screens, but the issue is that it breaks our consistency with other pages (connect and help)

**Comment by MaxLeiter** [comment]
Updated to @GewoonYorick's suggestion of nav on the side on desktop

image

image

**Comment by aab12345** [comment]
Side menu much better, need to see what it looks like on a mobile with there been less space to work with.

**Review by itsjohncs — CHANGES_REQUESTED** [review]
Haha the changed files view on GitHub scared me. Fortunately most of the lines have just been moved from one file to another. I used this to get a nicer diff:

```bash
git diff $(git merge-base HEAD master) --color-moved=plain --color-moved-ws=ignore-space-change
```

There's not much actual new/changed code in here which is great. I think we could probably spend some time polishing the styling (like on mobile there's not enough to contrast the tabs from the rest of the content so they're easy to miss), but I think it's worthwhile to just push this through and do any polishing as much-much smaller PRs.

I only found one thing that needs fixing. I didn't do super extensive testing on this but I'm hoping that since very little actual JS has changed we won't have any nasty surprises.

File: client/components/Settings/Navigation.vue [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	top: calc(45px + 5px);
+	/* Mid page minus width of container and 30 pixels for padding */
+	margin-left: calc(50% - 480px - 30px);
+}
+
+/** The calculation is mobile +  2/3 of container width. Fairly arbitrary. */
+@media screen and (max-width: calc(768px + 320px)) {
+	.settings-menu {
+		position: static;
+		width: 480px;
```
[the remark this answers is not in the document]
Comment (MaxLeiter) [reply client/components/Settings/Navigation.vue]: Fixed with `width: min(480px, 100%);`

**Review by itsjohncs — APPROVED** [review]
LGTM!

**Pull request merged**